src/docfx/build/manifest/ManifestBuilder.cs

```diff
@@ -21,21 +21,29 @@ public void MarkError(Document file)
             _filesWithErrors.Add(file);
         }
 
-        public bool TryAdd(Document file, FileManifest manifest)
+        public bool TryAdd(Document file, FileManifest manifest, List<string> monikers)
         {
             _manifest[file] = manifest;
 
             // TODO: see comments in Document.OutputPath.
             file.OutputPath = manifest.OutputPath;
 
             // Find publish url conflicts
-            if (!_filesBySiteUrl.TryAdd(manifest.SiteUrl, file))
+            if (!_filesBySiteUrl.TryAdd(manifest.SiteUrl, new ConcurrentBag<(Document doc, List<string> monikers)> { (file, monikers) })
+                && _filesBySiteUrl.TryGetValue(manifest.SiteUrl, out var existingFiles))
```

Comment:
this is also not thread-safe, just add to `_filesBySiteUrl` and to the conflict resolve in `Build` method.